lib/Models/ImageryLayerCatalogItem.js

```diff
@@ -831,75 +831,86 @@ ImageryLayerCatalogItem.disableLayer = function(catalogItem, layer, globeOrMap)
-        catalogItem._currentIntervalIndex = index;
-
-        if (preload) {
-            // Prefetch the (predicted) next layer.
-            // Here we use the terria clock because we want to optomise for the case where the item is playing on the
-            // timeline (which is linked to the terria clock) and preload the layer at the next time that the timeslider
-            // will move to.
-            var nextIndex = catalogItem.terria.clock.multiplier >= 0.0 ? index + 1 : index - 1;
-            if ((nextIndex === intervals.length) && (catalogItem.terria.clock.clockRange === ClockRange.LOOP_STOP)) {
-                nextIndex = 0;
-            }
-            if (nextIndex >= 0 && nextIndex < intervals.length && nextIndex !== catalogItem._nextIntervalIndex) {
-                var nextImageryProvider = catalogItem.createImageryProvider(catalogItem.intervals.get(nextIndex).data);
-                nextImageryProvider.enablePickFeatures = false;
-                catalogItem._nextLayer = ImageryLayerCatalogItem.enableLayer(catalogItem, nextImageryProvider, 0.0);
-                updateSplitDirection(catalogItem);
-                ImageryLayerCatalogItem.showLayer(catalogItem, catalogItem._nextLayer);
-                catalogItem._nextIntervalIndex = nextIndex;
-            }
+        clearNext();
+    }
+    // Given an interval index, set up an imagery provider and layer.
+    function nextLayerFromIndex(index) {
+        const imageryProvider = catalogItem.createImageryProvider(catalogItem.intervals.get(index).data);
+        imageryProvider.enablePickFeatures = false;
+        catalogItem._nextLayer = ImageryLayerCatalogItem.enableLayer(catalogItem, imageryProvider, 0.0);
+        updateSplitDirection(catalogItem);
+        ImageryLayerCatalogItem.showLayer(catalogItem, catalogItem._nextLayer);
+    }
+
+    const intervals = catalogItem.intervals;
+    if (!defined(currentTime) || !defined(intervals) || !catalogItem.isEnabled || !catalogItem.isShown) {
+        return;
+    }
+
+    const oldIndex = catalogItem._currentIntervalIndex;
+    if (oldIndex >= 0 && oldIndex < intervals.length && TimeInterval.contains(intervals.get(oldIndex), currentTime)) {
+        // the currently shown imagery's interval contains the requested time, so nothing to do.
+        return;
+    }
+    // Find the interval containing the current time.
+    const currentTimeIndex = intervals.indexOf(currentTime);
+    if (currentTimeIndex < 0) {
+        // No interval contains this time, so do not show imagery at this time.
+        ImageryLayerCatalogItem.disableLayer(catalogItem, catalogItem._imageryLayer);
+        clearCurrent();
+        return;
+    } else if (currentTimeIndex !== catalogItem._nextIntervalIndex) {
+        // We have a "next" layer, but it's not the right one, so discard it.
+        ImageryLayerCatalogItem.disableLayer(catalogItem, catalogItem._nextLayer);
+        // clearNext(); // seems redundant
```

Comment:
Oh, yeah. So the two lines above are replaced by `clearNext()`. But it seems redundant, because `replaceCurrentWithNext()` is about to be called, which calles `clearNext()` anyway.